frontend/src/metabase/components/ErrorPages/BugReportModal.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  Box,
+  Button,
+  Divider,
+  Flex,
+  Loader,
+  Modal,
+  Stack,
+  Text,
+} from "metabase/ui";
+
+import { DiagnosticCheckboxes } from "./DiagnosticCheckboxes";
+import type { ErrorPayload } from "./types";
+
+interface BugReportModalProps {
+  errorInfo: ErrorPayload;
+  onClose: () => void;
+  onSubmit: (values: any) => void;
+  onSlackSubmit: (values: any) => void;
+  canIncludeQueryData: boolean;
+  applicationName: string;
+  hiddenValues: Record<string, boolean>;
+  isSlackSending: boolean;
+}
+
+export const BugReportModal = ({
+  errorInfo,
+  onClose,
+  onSubmit,
+  onSlackSubmit,
+  canIncludeQueryData,
+  applicationName,
+  hiddenValues,
+  isSlackSending,
+}: BugReportModalProps) => {
+  const [isDetailsVisible, setIsDetailsVisible] = useState(false);
+
+  return (
+    <Modal opened onClose={onClose} title={t`Report a bug`} size={550}>
+      <FormProvider
+        initialValues={{
+          ...hiddenValues,
+          queryResults: false,
+          entityInfo: true,
+          frontendErrors: true,
+          backendErrors: true,
+          userLogs: true,
+          logs: true,
+          bugReportDetails: true,
+        }}
+        onSubmit={onSubmit}
+      >
+        {formik => (
+          <Form>
+            <Text py="md">
+              {t`What were you trying to do, and what steps did you take? What was the expected result, and what happened instead?`}
+            </Text>
+            <FormTextArea name="description" autoFocus />
+            <Box bg="#FAFAFB" p="lg" my="md" style={{ borderRadius: "0.5rem" }}>
+              <Flex align="flex-start" gap="md">
+                <Stack spacing="xs">
+                  <Text size="lg" weight="bold">
+                    {t`Include diagnostic information`}
+                  </Text>
+                  <Text color="text-medium" w="80%">
+                    {t`This information helps ${applicationName} figure out what exactly caused the issue.`}
+                  </Text>
+                </Stack>
+                <Button
+                  variant="subtle"
+                  p="0"
+                  fz="lg"
+                  miw="fit-content"
+                  onClick={() => setIsDetailsVisible(!isDetailsVisible)}
+                >
+                  {isDetailsVisible ? t`Done` : t`Edit`}
+                </Button>
+              </Flex>
+              {isDetailsVisible && (
+                <DiagnosticCheckboxes
+                  canIncludeQueryData={canIncludeQueryData}
+                  errorInfo={errorInfo}
+                  applicationName={applicationName}
+                />
+              )}
+            </Box>
+            <Button
+              variant="filled"
+              leftIcon={isSlackSending ? <Loader size="xs" /> : null}
+              onClick={() => {
+                trackSimpleEvent({
+                  event: "error_diagnostic_modal_submitted",
+                  event_detail: "submit-report",
+                });
```

Comment:
all use of `trackSimpleEvent` need to be a separate `analytics.ts` file. In the `ErrorPages` folder is fine.